provider/datagen/src/lib.rs

```diff
@@ -206,36 +222,179 @@ impl DatagenProvider {
+        let mut implicit = HashSet::new();
+        // TODO: Make including the default locale configurable
+        implicit.insert(DataLocale::default());
+        for locale in explicit.iter() {
+            let mut iter = fallbacker_with_config.fallback_for(locale.clone());
+            while !iter.get().is_empty() {
+                implicit.insert(iter.get().clone());
+                iter.step();
+            }
+        }
+
+        let result = supported_locales
+            .into_iter()
+            .chain(explicit.iter().cloned())
+            .filter(|locale| {
+                if implicit.contains(locale) {
+                    return true;
+                }
+                if explicit.contains(locale) {
+                    return true;
+                }
+                if locale.is_langid_und() && include_und {
+                    return true;
+                }
+                if locale.language().is_empty()
+                    && matches!(
+                        key.metadata().fallback_priority,
+                        icu_provider::FallbackPriority::Region
+                    )
+                {
+                    return true;
+                }
+                // Special case: skeletons *require* the -u-ca keyword, so don't export locales that don't have it
+                // This would get caught later on, but it makes datagen faster and quieter to catch it here
+                if key == icu_datetime::provider::calendar::DateSkeletonPatternsV1Marker::KEY
+                    && !locale.has_unicode_ext()
+                {
+                    return false;
+                }
+                let mut iter = fallbacker_with_config.fallback_for(locale.clone());
+                while !iter.get().is_empty() {
+                    if explicit.contains(iter.get()) {
+                        return true;
                     }
-                })
-                .collect(),
-            _ => unreachable!("resolved"),
+                    iter.step();
+                }
+                log::trace!("Filtered out: {key}/{locale}");
+                false
+            })
+            .collect();
+
+        Ok(result)
+    }
+
+    /// Loads a `DataPayload<ExportMarker>` with locale fallback enabled.
+    fn load_with_fallback(
+        &self,
+        key: DataKey,
+        locale: &DataLocale,
+    ) -> Result<Option<DataPayload<ExportMarker>>, DataError> {
+        log::trace!("Generating key/locale: {key}/{locale:}");
+        let mut metadata = DataRequestMetadata::default();
+        metadata.silent = true;
+        let mut config = LocaleFallbackConfig::from_key(key);
+        // Enable the `visit_default_script` option here, since some raw CLDR data needs it.
+        // However, we don't use it for evaluating runtime fallback, since this option is
+        // normally disabled at runtime.
+        config.visit_default_script = true;
```

Comment:
I don't think punching a `#[doc(hidden)]` hole in this API is a great solution. Can't `supported_locales`  return the correct normalization?

Reply:
The CLDR data is indexed as `sr-Latn-ME` and `zh-Hans-SG` so we need the fallback algorithm to hit those locales.

Reply:
Does CLDR data contain both `sr-Latn-ME` and `sr-ME`? If not, we should treat the `sr-Latn-ME.json` file as `sr-ME`.

Reply:
There is no `sr-ME` in CLDR, only `sr-Latn-ME`

Do you have a proposal for "we should treat the sr-Latn-ME.json file as sr-ME" which is better than the code I've written here to solve that problem?

Reply:
> There is no sr-ME in CLDR, only sr-Latn-ME

That seems weird, other tags are always minimised. What's the rationale for this?

> Do you have a proposal for "we should treat the sr-Latn-ME.json file as sr-ME" which is better than the code I've written here to solve that problem?

I haven't thought this through completely, but you can probably tweak the list in load functions in `CldrSource` (you'd have to give them a fallbacker or locale expander).